pkg/scheduler/api/devices/nvidia/gpushare/share.go

```diff
@@ -125,7 +125,7 @@ func checkNodeGPUSharingPredicate(pod *v1.Pod, gs *GPUDevices) (bool, error) {
 	if getGPUMemoryOfPod(pod) <= 0 {
 		return true, nil
 	}
-	ids := predicateGPUbyMemory(pod, gs)
+	ids := predicateGPUbyTotalMemory(pod, gs)
```

Comment:
I think this modification can solve the problem of resource filtering in gpuShare preemption mode. Does GPUNumber also need to do similar repairs?

Reply:
I believe that within the checkNodeGPUNumberPredicate function, the line:
`ids := predicateGPUbyNumber(pod, gs)`
should be replaced with:
`ids := predicateGPUbyTotalNumber(pod, gs)` //needs to be added

This is because the original function, predicateGPUbyNumber(pod, gs), utilizes allocatableGPUs := getDevicesIdleGPUs(gs).
As a result, if a node has all its GPU devices in use, it will not be considered for preemption at all.

If possible, I would like to address this issue in two separate steps and submit the subsequent fix in another pull request. This is because our cluster currently requires this solution.